ledger/sandbox-on-x/src/test/suite/scala/com/daml/ledger/sandbox/bridge/validate/ConflictCheckWithCommittedSpec.scala

```diff
@@ -0,0 +1,276 @@
+// Copyright (c) 2022 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.ledger.sandbox.bridge.validate
+
+import com.codahale.metrics.MetricRegistry
+import com.daml.error.{ContextualizedErrorLogger, DamlContextualizedErrorLogger}
+import com.daml.ledger.api.DeduplicationPeriod
+import com.daml.ledger.configuration.{Configuration, LedgerTimeModel}
+import com.daml.ledger.offset.Offset
+import com.daml.ledger.participant.state.index.v2.IndexService
+import com.daml.ledger.participant.state.v2.{SubmitterInfo, TransactionMeta}
+import com.daml.ledger.sandbox.bridge.validate.ConflictCheckWithCommittedSpec._
+import com.daml.ledger.sandbox.bridge.{BridgeMetrics, PreparedTransactionSubmission}
+import com.daml.ledger.sandbox.domain.Rejection._
+import com.daml.ledger.sandbox.domain.Submission
+import com.daml.lf.crypto.Hash
+import com.daml.lf.data.Time.Timestamp
+import com.daml.lf.data.{Ref, Time}
+import com.daml.lf.transaction.test.TransactionBuilder
+import com.daml.lf.transaction.{BlindingInfo, GlobalKey, Transaction}
+import com.daml.lf.value.Value
+import com.daml.lf.value.Value.ContractId
+import com.daml.logging.{ContextualizedLogger, LoggingContext}
+import com.daml.metrics.Metrics
+import com.daml.platform.apiserver.execution.MissingContracts
+import com.daml.platform.server.api.validation.ErrorFactories
+import org.mockito.{ArgumentMatchersSugar, MockitoSugar}
+import org.scalatest.Assertion
+import org.scalatest.flatspec.AsyncFlatSpec
+import org.scalatest.matchers.should.Matchers
+
+import java.time.Duration
+import scala.concurrent.Future
+
+class ConflictCheckWithCommittedSpec
+    extends AsyncFlatSpec
+    with Matchers
+    with MockitoSugar
+    with ArgumentMatchersSugar {
+
+  behavior of classOf[ConflictCheckWithCommittedImpl].getSimpleName
+
+  it should "validate causal monotonicity and key usages" in newTestContext { context =>
+    import context._
+    conflictCheckWithCommitted(input).map(_ shouldBe input)
+  }
+
+  it should "pass causal monotonicity check if referred contracts is empty" in newTestContext {
+    context =>
+      import context._
+
+      val submissionWithEmptyReferredContracts =
+        preparedTransactionSubmission.copy(inputContracts = Set(divulgedContract))
```

Comment:
```suggestion
        preparedTransactionSubmission.copy(inputContracts = inputContracts -- referredContracts)
```
Minor: describe the intent explicitly.

Reply:
Makes sense, thanks!